contracts/rewards/src/contract/execute.rs

```diff
@@ -30,6 +28,9 @@ where
         let stored_params = self.store.load_params();
         let epoch_duration: u64 = stored_params.params.epoch_duration.into();
         let epoch = stored_params.last_updated;
+        if cur_block_height < epoch.block_height_started {
```

Comment:
Shouldn't this be a panic situation?

Reply:
I feel like it's up to the caller to decide whether to panic or handle the error. But I don't feel strongly

Reply:
This one to me is the caller to blame, and later on may not be the only error that can occur in this method. I'd still prefer panic inside.